rllib/connectors/agent/state_buffer.py

```diff
@@ -8,21 +10,38 @@
     AgentConnector,
     ConnectorContext,
 )
+from ray import cloudpickle
 from ray.rllib.connectors.registry import register_connector
 from ray.rllib.policy.sample_batch import SampleBatch
 from ray.rllib.utils.spaces.space_utils import get_base_struct_from_space
 from ray.rllib.utils.typing import ActionConnectorDataType, AgentConnectorDataType
 from ray.util.annotations import PublicAPI
 
 
+logger = logging.getLogger(__name__)
+
+
 @PublicAPI(stability="alpha")
 class StateBufferConnector(AgentConnector):
-    def __init__(self, ctx: ConnectorContext):
+    def __init__(self, ctx: ConnectorContext, states: Any = None):
         super().__init__(ctx)
 
         self._initial_states = ctx.initial_states
         self._action_space_struct = get_base_struct_from_space(ctx.action_space)
+
         self._states = defaultdict(lambda: defaultdict(lambda: (None, None, None)))
+        if states:
+            try:
+                self._states = cloudpickle.loads(states)
+            except pickle.UnpicklingError:
+                # StateBufferConnector states are only needed for rare cases
+                # like stashing then restoring a policy during the rollout of
+                # a single episode.
+                # It is ok to ignore the error for most of the cases here.
+                logger.info(
```

Comment:
I don't get why you should not error out all the time here? When will you ever pass in a states object in that is ok if it's not unpickled?

Reply:
when you recover a policy for serving, we wouldn't need the state buffer state.
actually this comment reminded me, I am letting StateBufferConnector to clear state whenever someone switch it into eval mode.

Reply:
In that case we should not pass in the states at all?

Reply:
maybe we can discuss a bit.
I feel like the only case we need to restore this is when policies are stashed and recovered in the middle of an episode.
we don't need this state even for training as long as we keep the policy in cache throughout an episode.
so maybe a better fix is to ping any "in-use" poclies, and prevent them from being stashed.